pilot/cmd/pilot-discovery/main.go

```diff
@@ -152,6 +152,9 @@ func init() {
 	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableCaching, "discoveryCache", true,
 		"Enable caching discovery service responses")
 
+	discoveryCmd.PersistentFlags().StringVar(&serverArgs.MeshUID, "meshUID", "",
```

Comment:
No more flags - use env variable, not friendly to kustomize and new installer.

Once viper PR is merged will be less of an issue since all settings will be env or file-based and we can kill the CLI.

Reply:
Also: since this is coming from mesh config, I think we should just load it directly from file - again, viper can handle re-loading but we can just load it as static.

Reply:
@costinm Are you suggesting this one should be handled differently from the 20+ others above this? Wouldn't that make more sense as part of a larger refactoring?